docs/pages/application-access/cloud-apis/google-cloud.mdx

```diff
@@ -0,0 +1,615 @@
+Logged into GCP app "google-cloud-cli".
+Your service account: teleport-google-cloud-cli@my-project.iam.gserviceaccount.com
+Example command: tsh gcloud compute instances list
+```
+
+### Execute Google Cloud CLI commands
+
+At this point, you can run `gcloud` commands using the Teleport Application
+Service by prefixing them with `tsh`. Since your user authenticated to your
+Google Cloud CLI application with a service account that can list VMs, for
+example, run this command to do so: 
+
+```code
+$ tsh gcloud compute instances list
+```
+
+You should see a list of virtual machines in your Google Cloud project.
+
+However, your Teleport user cannot create a VM, since its service account
+does not have this authorization:
+
+```code
+$ tsh gcloud compute instances create another-instance --zone=<Var name="google-cloud-zone" />
+ERROR: (gcloud.compute.instances.create) Could not fetch resource:
+ - Required 'compute.instances.create' permission for 'projects/my-project/zones/my-zone/instances/another-instance'
+
+ERROR: exit status 1
+```
+
+### Use Google Cloud CLI applications without `tsh`
+
+In addition to running `gcloud` commands via `tsh`, you can grant secure access
+to any CLI application that executes commands against Google Cloud's APIs.
+
+To do this, use `tsh` to start a local proxy that forwards traffic from your CLI
+application to the Teleport Application Service. The Application Service uses
+the `teleport-google-cloud-cli` service account we created earlier to fetch an
+authentication token from Google Cloud. Your CLI application uses this token to
+authenticate requests to Google Cloud's APIs.
+
+To start the local proxy, run the following `tsh` command:
+
+```code
+$ tsh proxy gcloud 
+```
+
+The command will print the address of the local proxy server along with
+`export` commands for assigning environment variables. Google Cloud CLI
+applications read these variables in order to request an authentication token
+for Google Cloud's APIs:
+
+```text
+Started GCP proxy on http://127.0.0.1:50614.
+To avoid port randomization, you can choose the listening port using the --port flag.
+
+Use the following credentials and HTTPS proxy setting to connect to the proxy:
+
+  export CLOUDSDK_AUTH_ACCESS_TOKEN=00000000000000000000000000000000
+  export CLOUDSDK_CORE_CUSTOM_CA_CERTS_FILE=/Users/myuser/.tsh/keys/teleport.example.com/myuser-app/teleport.example.com/google-cloud-cli-localca.pem
+  export HTTPS_PROXY=http://127.0.0.1:50614
+```
+
+<Notice type="warning">
+
+`tsh proxy gcloud` runs the local proxy in the foreground, so don't interrupt
+the process or exit the terminal where you ran the command until you're ready
+to close the local proxy.
+
+</Notice>
+
+Copy the `export` commands and paste them into a second terminal. In that
+terminal, you can now run your Google Cloud CLI application of choice. For
+example, you can run the following command to list Google Compute Engine VMs:
+
+```code
+$ gcloud compute instances list
+```
+
```

Comment:
`gsutil ls` would work in this shell now too, but perhaps this is redundant to point out here.